### PR TITLE
DTSPB-3541 hmrc to blob

### PR DESCRIPTION
**Pull request by DelythJustice**
### JIRA link (if applicable) ###
DTSPB-3541

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -131,7 +131,8 @@ dependencies {
 ./gradle bootWithCCD 
 # run in debugger with 
 ./gradlew bootWithCCD --debug-jvm
-or just click debug in intellij on probate-back-office [bootWithCCD] run/debug configuration
```
Comment (FeliTam) [inline README.md]: mistyped

File: src/contractTest/resources/application.yml [hunk 2]
```diff
@@ -465,6 +468,7 @@ data-extract:
     size: 100
 
 smeeAndFordEmail: ${SMEE_AND_FORD_EMAIL:smeeAndFord@probate-test.com}
+hmrcEmail: ${HMRC_EMAIL:probate-hmrc-extract-aaaakmsw7ddsfspzaz5f3upafi@moj.org.slack.com}
```
Comment (FeliTam) [inline src/contractTest/resources/application.yml]: think we better leave a dummy email here in actual release

File: src/integrationTest/resources/application.yml [hunk 3]
```diff
@@ -399,6 +402,7 @@ idam-s2s-auth:
 
 excelaEmail: exela@probate-test.com
 smeeAndFordEmail: smeeAndFord@probate-test.com
+hmrcEmail: ${HMRC_EMAIL:probate-hmrc-extract-aaaakmsw7ddsfspzaz5f3upafi@moj.org.slack.com}
```
Comment (FeliTam) [inline src/integrationTest/resources/application.yml]: better leave a dummy email here as well

File: src/main/java/uk/gov/hmcts/probate/service/dataextract/HmrcDataExtractService.java [hunk 4]
```diff
@@ -46,6 +51,22 @@ private void performHmrcExtract(String date) {
         uploadHmrcFile(date, date, casesFound);
```
Comment (FeliTam) [inline src/main/java/uk/gov/hmcts/probate/service/dataextract/HmrcDataExtractService.java:57]: Need to change to emailHmrcFile as well

File: src/main/java/uk/gov/hmcts/probate/service/EmailWithFileService.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        try {
+            fileContents = FileUtils.readFileToByteArray(file);
+        } catch (IOException e) {
+            log.error("Error reading HMRC file {}", e.getMessage());
+            return false;
+        }
+
+        HashMap<String, Object> personalisation = new HashMap<>();
+        if (prepareUpload(fileContents, personalisation)) {
+            return false;
```
Comment (FeliTam) [inline src/main/java/uk/gov/hmcts/probate/service/EmailWithFileService.java]: When getting NotificationClientException in prepareUpload, it returns false.
Then in here should be
`if (!prepareUpload(fileContents, personalisation))`

File: README.md (hunk 1 above)
Comment (DelythJustice) [reply README.md]: will undo my accidental change on readme

File: src/main/java/uk/gov/hmcts/probate/service/EmailWithFileService.java (hunk 5 above)
Comment (DelythJustice) [reply src/main/java/uk/gov/hmcts/probate/service/EmailWithFileService.java]: yeh, noticed this mistake when testing too

**Review by FeliTam — APPROVED** [review]

**Pull request merged**